dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/XPathMapper.java

```diff
@@ -88,27 +85,54 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
       xpw.setDefaultNamespace(binding.defaultNamespace());
     }
 
-    // Resolve variables, if variables and valueTemplate are set
+    // Resolve variables, if variables and valueTemplate are set. Otherwise, evalute expressions
+    boolean multiLanguage = method.getReturnType().isAssignableFrom(Map.class);
     if (!isEmptyOrBlankStringSet(variables) && !binding.valueTemplate().isEmpty()) {
-      return evaluteVariablesAndTemplate(binding, binding.valueTemplate(), variables);
+      return evaluteVariablesAndTemplate(binding, variables, multiLanguage);
     } else {
-      return evaluateExpressions(binding, expressions);
+      boolean multiValue;
+      if (multiLanguage) {
+        multiValue = method.getGenericReturnType().getTypeName()
+            .equals(RETURN_TYPE_MULTI_LANGUAGE_MULTI_VALUE);
```

Comment:
Ich habe gerade nochmal etwas recherchiert und ich glaube es geht eleganter (und typsicherer) wenn du `instanceof` checks auf dem `Type` Objekt machst das von `getGenericReturnType()` zurückgegeben wird.

Einstiegspunkt in der JDK Doku: https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Type.html

Beispiel für `Map<Locale, String>`, ohne Garantie auf Richtigkeit:

```java
Type genericType = method.getGenericReturnType();
multiValue =
    genericType instanceof ParameterizedType
    && List.class.isAssignableFrom(
       (Class<?>) ((ParameterizedType) genericType).getActualTypeArguments[1]);
```

Die Java Reflection API ist wirklich wirklich herrlich :-/

Reply:
Man könnte an der Stelle glaube ich auch wirklich gut vertreten eine Guava Dependency anzuziehen, die haben nämlich eine API die das etwas angenehmer macht: https://github.com/google/guava/wiki/ReflectionExplained

Reply:
Ist jetzt umgebaut und damit wesentlich eleganter und sicherer